aws/amazon-chime-sdk-js
# Added Android WebView support

**Issue #:**
Implement Android Webview Support
**Description of changes:**
- Added Android WebView support
- Added a fix in the browser demo in cases where the browser does not support grabbing device labels. The key issue is that the browser demo will attempt to go into flow-need-permissions and flow-devices whenever enumerateDevices is called and returns empty. In Android WebView enumerateDevices does not work. More info can be found here: https://bugs.chromium.org/p/chromium/issues/detail?id=669492

**Testing**
1. Have you successfully run `npm run build:release` locally?
Yes
2. How did you test these changes?
- Build the sdk
- Build the browser demo
- Deploy the serverless demo and copy the given url for later
- Clone the AndroidWebview branch from https://github.com/aws-samples/amazon-chime-sdk
- Launch the newly added Android-Webview-sample folder in Android Studio 4.2.2
- In the AppConfig.kt file replace PLACEHOLDER with the url from the serverless demo
- Connect a physical android device that is in developer mode
- Launch the app on the physical device
- On the android device press the 'Go to WebView' button
- Fill in the fields on the page that appears
- Click continue
- Click continue on the next page
- Open the serverless url on another device and browser, then perform the last 3 steps
- Test for audio input on the android device
- Test for audio output on the android device
- Test for video input on the android device
3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it?
Yes, can be tested using the serverless demos
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
Added a new browserbehavior function called doesNotSupportDevicesLabels,  this has been reviewed and approved.
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
No
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
